### PR TITLE
shard_map: add API docs

### PR DESCRIPTION
**Pull request by jakevdp**

### CODE REVIEW

**Comment by yashk2810** [comment]
I think @mattjj also had a sketch of the docs somewhere when he was trying to take it out of experimental. Maybe you want to merge your and his version?

**Comment by 8bitmp3** [comment]
SGTM @yashk2810 @jakevdp @mattjj 

Right now, as discussed with @jakevdp, there are no visible API docs, no docstrings for `shard_map`. But we have docs/notebooks.

**Comment by jakevdp** [comment]
Sure, the main thing is to have *something* documented. It looks like my docstring and Matt's are pretty similar.

**Comment by jakevdp** [comment]
I pulled in some of the docstring verbiage from #20124

File: jax/experimental/shard_map.py [hunk 1]
```diff
@@ -82,6 +80,56 @@
 @traceback_util.api_boundary
 def shard_map(f: Callable, mesh: Mesh, in_specs: Specs, out_specs: Specs,
               check_rep: bool = True, auto: frozenset[AxisName] = frozenset()):
+  """Map a computation across shards of data.
```
Comment (mattjj) [inline jax/experimental/shard_map.py:83]:
```suggestion
  """Map a function over shards of data.
```

**Review by mattjj — APPROVED** [review]
Thanks!

File: jax/experimental/shard_map.py (hunk 1 above)
Comment (jakevdp) [reply jax/experimental/shard_map.py]: Done, thanks!

**Pull request merged**